modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/MultiStatementHandler.java

```diff
@@ -239,6 +240,16 @@ CompletableFuture<AsyncSqlCursor<InternalSqlRow>> processNext() {
                             });
                 }
             });
+        } catch (TxControlInsideExternalTxNotSupportedException txEx) {
```

Comment:
It looks like this fix fixes the problem with handling `TxControlInsideExternalTxNotSupportedException` (but we still need to open a ticket to think about what to do with similar cases, for example select ..; insert .. 1/0).

If it is decided that this fix is ​​suitable (for me it looks good), I would recommend changing the test so that it immediately fails if the behavior changes.

e.g. instead of "SELECT 1; COMMIT"
read several pages

    boolean res = stmt.execute("SELECT x FROM TABLE(SYSTEM_RANGE(0, 10000));COMMIT");
    assertTrue(res);
    assertNotNull(stmt.getResultSet());
    try (ResultSet rs = stmt.getResultSet()) {
        while (rs.next()) {
            System.out.println(rs.getInt(1));
        }
    }